node/src/Pos/Client/Txp/Balances.hs

```diff
@@ -53,17 +58,31 @@ type BalancesEnv ext ctx m =
     , MonadMask m
     , MonadTxpMem ext ctx m)
 
+-- TODO [CSM-407] Move it somewhere :/
+-- | Get active pending transactions which are *not* in blockchain yet.
+getRecentTxs :: WebWalletModeDB ctx m => m [WithHash Tx]
+getRecentTxs = tryTopsortTxs . mapMaybe extract <$> WS.getPendingTxs
+  where
+    extract PendingTx{..} = do
+        let PtxApplying th = _ptxCond
+        Just $ WithHash (_thTx th) _ptxTxId
+    tryTopsortTxs txs = fromMaybe txs $ topsortTxs identity txs
```

Comment:
I think that it's not okay to just silently return unsorted tx list if we couldn't topsort list of transactions.
I'm not sure how exactly `topsortTxs` works now, but I believe that it should support transactions which depend on same `TxIn`s appearing in the list (which might perfectly be the case for pending pool). If it's so, then we clearly should throw an error if topsort fails - because it means that transactions in pending pool form a loop, which is nonsense.
If `topsortTxs` doesn't support such transactions (which is not likely), then I think it itself should be upgraded to support them.